include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
-                // (a * 10^x) / (b*10^y) = (a*10 / b*10) * 10^((x-1)-(y-1))
-                // 100/20 -> 1/2 * (10)
-                int exponent_dif = (numerator.exponent - 1) - (divisor.exponent - 1);
+                /* Special Cases */
+                if((*this) == zero){
+                    return;
+                }
 
-                numerator.exponent = 1;
-                divisor.exponent = 1;
-                tmp.digits = {1};
-                tmp.exponent = 1;
+                exact_number<T> one;
+                one.digits={1};
+                one.exponent=1;
+                if(divisor == one){
+                    return;
+                }
 
-                exact_number<T> zero = exact_number<T>(); 
+                bool positive = (this->positive == divisor.positive);
 
-                if (divisor == tmp) {
-                    this->exponent = exponent_dif + 1;
+                exact_number<T> minus_one = one;
+                minus_one.positive = false;
+                if(divisor == minus_one){
                     this->positive = positive;
                     return;
                 }
 
-                if (divisor == (*this)) { 
-                    (*this) = tmp;
-                    (*this).positive = positive;
+                if(divisor == (*this)){
+                    (*this) = one;
                     return;
                 }
 
-                if ((*this) == zero)
-                    return;
-                    
-                if (divisor == zero)
-                    throw(boost::real::divide_by_zero());
-
-                // N < D --> 0 < abs(Q) < 1
-                if (numerator < divisor) {
-                        left = exact_number<T>(); // 0
-                        right = tmp; // 1
-                    } else { // assuming D > 1. N > D ---> 1 < N / D < N
-                        left = tmp; // 1
-                        right = numerator;
-                    }
+                const T base = (std::numeric_limits<T>::max() / 4) * 2 - 1;
+                exact_number<T> half;
+                half.digits = {base / 2 + 1};
+
+                exact_number<T> left, right, numerator, denominator, residual, length;
+                numerator = (*this);
+                denominator = divisor;
+                numerator = numerator.abs();
+                denominator = denominator.abs();
 
-                // distance = (right - left) / 2
-                distance = (right - left) * half;
-                (*this) = left + distance;
-                // N/D = Q -> QD -N = 0
-                residual = (*this) * divisor - numerator;
-                if (residual == zero) {
-                    this->exponent += exponent_dif;
+                int exponent_diff = (this->exponent - 1) - (denominator.exponent - 1);
+                numerator.exponent = 1;
+                denominator.exponent = 1;
+                
+
+                if(numerator > denominator){
+                    left = one;
```

Comment:
What is right and left? 
We can really use some comments on this area of the code to try to understand what you trying to do.
When you write them, think about someone reading this in 5 years after a bug report was filled trygin to figure out what is going on. It can be you, or anyone else.

Reply:
Yes, I will add comments